captum/attr/_core/lime.py

```diff
@@ -0,0 +1,1007 @@
+                    the interpretable representation with either
+                    to_interp_rep_transform or from_interp_rep_transform.
+            from_interp_rep_transform (callable): Function which takes a
+                    single sample's interpretable representation (tensor
+                    of shape 1 x num_interp_features) and returns
+                    the corresponding input in the original input space
+                    (mathing shapes of original input to attribute).
+
+                    This argument is necessary if sample_interpretable_space
+                    is True, otherwise None can be provided for this argument.
+
+                    The expected signature of this callable is:
+
+                    from_interp_rep_transform(
+                        curr_sample: Tensor [2D 1 x num_interp_features]
+                        original_input: Tensor or Tuple of Tensors,
+                        **kwargs: Any
+                    ) -> Tensor or tuple of Tensors
+
+                    Returned sampled input should match the type of original_input
+                    and corresponding tensor shapes.
+
+                    Note that kwargs include all kwargs passed to the attribute
+                    method.
+
+            to_interp_rep_transform (callable): Function which takes a
+                    sample in the original input space and converts to
+                    its interpretable representation (tensor
+                    of shape 1 x num_interp_features) and returns
+                    the corresponding input in the original input space.
+
+                    This argument is necessary if sample_interpretable_space
+                    is False, otherwise None can be provided for this argument.
+
+                    The expected signature of this callable is:
+
+                    to_interp_rep_transform(
+                        curr_sample: Tensor or Tuple of Tensors,
+                        original_input: Tensor or Tuple of Tensors,
+                        **kwargs: Any
+                    ) -> Tensor [2D 1 x num_interp_features]
+
+                    curr_sample will match the type of original_input
+                    and corresponding tensor shapes.
+
+                    Note that kwargs include all kwargs passed to the attribute
+                    method.
+        """
+        PerturbationAttribution.__init__(self, forward_func)
+        self.train_interpretable_model_func = train_interpretable_model_func
+        self.similarity_func = similarity_func
+        self.sampling_func = sampling_func
+        self.sample_interpretable_space = sample_interpretable_space
+        self.from_interp_rep_transform = from_interp_rep_transform
+        self.to_interp_rep_transform = to_interp_rep_transform
+
+        if self.sample_interpretable_space:
+            assert (
+                self.from_interp_rep_transform is not None
```

Comment:
Yes, we still need this, because we have to convert the interpretable representation to an input to the original model in order to obtain the model's output.

Reply:
Right - of course. Thanks for clarification.